crates/re_viewport/src/auto_layout.rs

```diff
@@ -60,234 +33,86 @@ pub(crate) fn tree_from_space_views(
-            });
-
-            SpaceMakeInfo {
-                id: *space_view_id,
-                path: space_view.space_origin.clone(),
-                category: space_view.category,
-                aspect_ratio,
-            }
+        .map(|(space_view_id, space_view)| SpaceMakeInfo {
+            id: *space_view_id,
+            category: space_view.category,
         })
         .collect_vec();
 
-    if space_make_infos.is_empty() {
-        egui_tiles::Tree::empty()
+    let mut tiles = egui_tiles::Tiles::default();
+
+    let root = if space_make_infos.len() == 1 {
+        tiles.insert_pane(space_make_infos[0].id)
+    } else if space_make_infos.len() == 3 {
+        // Special-case for common case that doesn't fit nicely in a grid
+        arrange_three(
+            [
+                space_make_infos[0].clone(),
+                space_make_infos[1].clone(),
+                space_make_infos[2].clone(),
+            ],
+            &mut tiles,
+        )
     } else {
-        let mut tiles = egui_tiles::Tiles::default();
-        // Users often organize by path prefix, so we start by splitting along that
-        let layout = layout_by_path_prefix(viewport_size, &mut space_make_infos);
-        let root = tree_from_split(&mut tiles, &layout);
-        egui_tiles::Tree::new(root, tiles)
-    }
+        // Arrange it all in a grid that is responsive to changes in viewport size:
+        let child_tile_ids = space_make_infos
+            .into_iter()
+            .map(|smi| tiles.insert_pane(smi.id))
+            .collect_vec();
+        tiles.insert_grid_tile(child_tile_ids)
+    };
+
+    egui_tiles::Tree::new(root, tiles)
 }
 
-fn tree_from_split(
+fn arrange_three(
+    mut spaces: [SpaceMakeInfo; 3],
     tiles: &mut egui_tiles::Tiles<SpaceViewId>,
-    split: &LayoutSplit,
 ) -> egui_tiles::TileId {
-    match split {
-        LayoutSplit::LeftRight(left, fraction, right) => {
-            let container = egui_tiles::Linear::new_binary(
-                egui_tiles::LinearDir::Horizontal,
-                [tree_from_split(tiles, left), tree_from_split(tiles, right)],
-                *fraction,
-            );
-            tiles.insert_container(container)
+    // We will arrange it like so:
+    //
+    // +-------------+
+    // |             |
+    // |             |
+    // |             |
+    // +-------+-----+
+    // |       |     |
+    // |       |     |
+    // +-------+-----+
+    //
+    // or like so:
+    //
+    // +-----------------------+
+    // |          |            |
+    // |          |            |
+    // |          +------------+
+    // |          |            |
+    // |          |            |
+    // |          |            |
+    // +----------+------------+
+    //
+    // But which space gets a full side, and which doesn't?
+    // Answer: we prioritize them by category:
+
+    /// lower is better
+    fn category_priority(category: ViewCategory) -> usize {
```

Comment:
quite unhappy to see this new use of `ViewCategory` when I'm still trying to get rid of it (not finding the time between all the higher prioritized things). Can you instead introduce a field on `SpaceViewClass` and/or replace the aspect ratio field we have there?

Reply:
I want to keep aspect ratio for future use (when integrated into egui_tiles at some theoretical point in time), but yeah, I can add something like `LayoutPriority` or something

Reply:
Fixed!